core/property/registry.go

```diff
@@ -84,10 +84,13 @@ func getValueFromResolver(prop interface{}) (interface{}, bool) {
 // expressions, the resolved value is returned and the boolean is true.
 // Otherwise the returned value will be nil and the boolean will be false.
 
-func Resolve(name string) (interface{}, bool) {
+func Resolve(value interface{}) (interface{}, bool) {
```

Comment:
why are we resolving non string? Sorry, a method with input and output interface type gives no information to the user of what is valid whatsoever.